martin/build.rs

```diff
@@ -1,13 +1,115 @@
-fn main() -> std::io::Result<()> {
-    #[cfg(feature = "webui")]
-    {
-        static_files::NpmBuild::new("martin-ui")
-            .install()?
-            .run("build")?
-            .target("martin-ui/dist")
-            .change_detection()
-            .to_resource_dir()
-            .build()?;
+#[cfg(feature = "webui")]
+use std::fs;
+#[cfg(feature = "webui")]
+use std::path::Path;
+
+#[cfg(feature = "webui")]
+fn copy_file_tree(src: &Path, dst: &Path, exclude_dirs: &[&str]) {
+    assert!(
+        src.is_dir(),
+        "source for the copy operation is not an existing directory"
+    );
+    let _ = fs::remove_dir_all(dst); // ignore if dir does not exist
+    fs::create_dir_all(dst).unwrap_or_else(|e| {
+        panic!(
+            "failed to create destination directory {}: {e}",
+            dst.display()
+        )
+    });
+    let excludes = exclude_dirs.iter().map(|v| src.join(v)).collect::<Vec<_>>();
+
+    let mut it = walkdir::WalkDir::new(src).into_iter();
```

Comment:
> Crate walkdir provides an efficient and cross platform implementation of recursive directory traversal. ***Several options are exposed to control iteration, such as whether to follow symbolic links (default off)***, limit the maximum number of simultaneous open file descriptors and the ability to efficiently skip descending into directories

Something to be aware of => no directories which are symlinks

We could disable this footgun.. What do you think?

```suggestion
    let mut it = walkdir::WalkDir::new(src).follow_links(true).into_iter();
```

Reply:
you mean just in case we start using symlinks?  Sure